cleverpush/src/main/java/com/cleverpush/banner/AppBannerModule.java

```diff
@@ -725,15 +729,91 @@ private boolean checkEventTriggerCondition(BannerTriggerCondition condition) {
     return false;
   }
 
+  private boolean checkTargetEventRelationFilter(CheckFilterRelation relation, String compareValue,
+                                                 String attributeValue, String fromValue, String toValue, String bannerId, String event) {
+    try {
+      if (events.isEmpty() || event == null || event.isEmpty()) {
+        return false;
+      }
+
+      for (TriggeredEvent triggeredEvent : events) {
+        if (triggeredEvent.getId() == null || !triggeredEvent.getId().equals(event) || compareValue == null || compareValue.isEmpty()) {
+          return false;
+        }
+
+        List<TableBannerTrackEvent> bannerTrackEvents = DatabaseClient.getInstance(CleverPush.context)
+                .getAppDatabase()
+                .trackEventDao()
+                .getBannerTrackEvent(bannerId, event, compareValue, attributeValue, String.valueOf(relation), fromValue, toValue);
+
+        if (bannerTrackEvents.isEmpty()) {
+          return false;
+        }
+
+        boolean targetEvents = false;
```

Comment:
you can do it a bit simpler, but you dont need to
you can remove targetEvents and conditionTrue
and instead of
`
if (conditionTrue) {
            targetEvents = true;
            break;
          }
`
you do 
`
return true;
`

and in the switch case you do like this
`
if (count != Integer.parseInt(attributeValue)) {
                continue;
              }
`
and beofre you also do
`
if (pastDays > Integer.parseInt(property)) {
            continue;
          }
`